### ISSUE TITLE
`run_conversion` fails to add data when in-memory nwbfile is passed. Feature or bug?

### ISSUE THREAD

**Issue by h-mayorquin** [catalystneuro/neuroconv#1159]
Summary: when using `converter.run_conversion` and the `nwbfile` argument is used to pass an in-memory nwbfile then the data from the converter is not added. This sounds like a bug to me but I don't know if this was intended behavior at some point. 

What do you think? @pauladkisson @weiglszonja @bendichter @alessandratrapani 

The reason for this behavior is found on these lines that only call `add_to_nwbfile` if no nwbfile is provided.

https://github.com/catalystneuro/neuroconv/blob/05423d16f0d55baf6b2b2679f89eddc881a6581c/src/neuroconv/nwbconverter.py#L272-L274

For a concrete example, see the following:

```python
from neuroconv.tools.testing.mock_interfaces import MockRecordingInterface, MockSortingInterface, MockImagingInterface
from neuroconv import ConverterPipe
from tempfile import TemporaryDirectory



imaging_interface = MockImagingInterface()
recording_interface = MockRecordingInterface()
sorting_interface = MockSortingInterface()

nwbfile = imaging_interface.create_nwbfile()

converter = ConverterPipe(data_interfaces=[recording_interface, sorting_interface])

with TemporaryDirectory() as tmpdir:
    tmp_path = Path(tmpdir)
    nwbfile_path = tmp_path / "test_conversion.nwb"
    converter.run_conversion(nwbfile_path=nwbfile_path, nwbfile=nwbfile)
    
    from pynwb import NWBHDF5IO 
    with NWBHDF5IO(str(nwbfile_path), 'r') as io:
        nwbfile = io.read()
        assert nwbfile.units is not None
        assert len(nwbfile.aquisition) > 1
        

```
Both assertions fail no matter the order. In other words, neither the recording interface nor sorting interface data are added when calling run conversion. 

**Relevant**: the existence of this behavior is used in our testing here:

https://github.com/catalystneuro/neuroconv/blob/05423d16f0d55baf6b2b2679f89eddc881a6581c/src/neuroconv/tools/testing/data_interface_mixins.py#L227-L238

Where the **test would fail if null behavior were to change** as the converter would try to add the data again.

**Comment by pauladkisson** [comment]
Seems like a bug to me.

**Comment by weiglszonja** [comment]
I agree with @pauladkisson, the intended behavior _should_ be to write data from the converter and the in-memory nwbfile. Maybe it has something to do with append mode that it was designed this way?

**Comment by h-mayorquin** [comment]
OK, I also think this is a bug. Will provide a fix.
